cmd/goal/box.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"bytes"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var boxName string
+var maxBoxes uint64
+
+func init() {
+	appCmd.AddCommand(appBoxCmd)
+
+	appBoxCmd.AddCommand(appBoxInfoCmd)
+	appBoxCmd.AddCommand(appBoxListCmd)
+	appBoxCmd.PersistentFlags().Uint64Var(&appIdx, "app-id", 0, "Application ID")
+	appBoxCmd.MarkFlagRequired("app-id")
+
+	appBoxInfoCmd.Flags().StringVarP(&boxName, "name", "n", "", "Application box name. Use the same form as app-arg to name the box.")
```

Comment:
I'm willing to do if the weight of opinion is in the direction. I did consider it, but I preferred to have a single source of truth, so that we don't forget to update all the documentation points if/as we add new forms. As an example of the kind of mistake that repetition creates, this is not quite right:

```
        "description": "Given an application ID and box name, it returns the box name and value (each base64 encoded). Box names must be in the goal app call arg encoding form 'encoding:value'. For ints, use the form 'int:1234'. For raw bytes, use the form 'b64:A=='. For printable strings, use the form 'str:hello'. For addresses, use the form 'addr:XYZ...'.",
```

because it implies that it is exhaustive.  I'd much rather it stopped after the "goal app arg" sentence.  This doesn't mention ABI types or b32 encoding, (or "base64", "base32", "string", "integer", "address" - all are allowed!).

Now, of course, the `app-arg` definition _also_ doesn't mention these other prefixes.  But if/when we fix it, it seems bad to track down several more instances.  But I totally get that there is a downside, a user might be annoying by needing to track this info down.

Reply:
Could this be solved by a shares message variable?